### PR TITLE
AV-203812: Add doc wrt EnableEndpointSlices value

### PR DESCRIPTION
**Pull request by ashishnayak96**

### CODE REVIEW

**Review by aaha97 — COMMENTED** [review]
LGTM

**Review by aaha97 — DISMISSED** [review]

File: docs/values.md [hunk 1]
```diff
@@ -342,4 +342,8 @@ Enable Gateway API in the featureGate to use this field.
 
 ### GatewayAPI.image.repository
 
-If you are using a private container registry and you'd like to override the default dockerhub settings, then this field can be edited with the private registry name.
\ No newline at end of file
+If you are using a private container registry and you'd like to override the default dockerhub settings, then this field can be edited with the private registry name.
+
+### featureGates.EnableEndpointSlice
+
+Enable this flag to use EndpointSlice instead of Endpoints in AKO. This also supports graceful shutdown of servers. Enabled by default from AKO 1.13.1.
```
Comment (pkoshtavmware) [inline docs/values.md:349]: EndpointSlice -> EndpointSlices

Comment (ashishnayak96) [reply docs/values.md]: done

**Review by pkoshtavmware — DISMISSED** [review]
LGTM

**Review by akshayhavile — COMMENTED** [review]
@ashishnayak96: We should add documentation in file `docs/install/helm.md` as it defines parameter that can be configured during helm install command. I think we are missing `FeatureGates.gatewayAPI` so we should include that also.

Also it is not part of your PR but in existing hostrule.md file in configure-tcp-settings section (https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/blob/master/docs/crds/hostrule.md#configure-tcp-settings), `tcpSetting` should be `tcpSettings` in example mentioned in the section. Can you please do this also?

**Review by akshayhavile — APPROVED** [review]

File: docs/install/helm.md [hunk 2]
```diff
@@ -151,6 +151,10 @@ The following table lists the configurable parameters of the AKO chart and their
 | `NetworkSettings.enableRHI` | Publish route information to BGP peers | false |
 | `NetworkSettings.bgpPeerLabels` | Select BGP peers using bgpPeerLabels, for selective VsVip advertisement. | `Empty List` |
 | `NetworkSettings.nsxtT1LR` | Unique ID (note: not display name) of the T1 Logical Router for Service Engine connectivity. Only applies to NSX-T cloud.| `Empty string` |
+| `FeatureGates.gatewayAPI` | FeatureGates is to enable or disable experimental features. GatewayAPI feature gate enables/disables processing of Kubernetes Gateway API CRDs. | false |
+| `FeatureGates.enableEndpointSlice` | EnableEndpointSlice feature gate enables/disables processing of Kubernetes EndpointSlices instead of Endpoints. | true |
+| `GatewayAPI.Image.repository` | Specify docker-registry that has the ako-gateway-api image | projects.registry.vmware.com/ako/ako-gateway-api:1.11.1 |
```
[the remark this answers is not in the document]
Comment (ashishnayak96) [reply docs/install/helm.md]: There is no version in default value. Removing it

File: docs/ako_operator.md [hunk 3]
```diff
@@ -123,6 +123,7 @@ The following table also lists the configurable fields in the `AKOConfig` object
 | `NetworkSettings.bgpPeerLabels` | Select BGP peers using bgpPeerLabels, for selective VsVip advertisement. | `Empty List` |
 | `NetworkSettings.nsxtT1LR` | Specify the T1 router for data backend network, applicable only for NSX-T based deployments| `Empty string` |
 | `FeatureGates.gatewayAPI` | FeatureGates is to enable or disable experimental features. GatewayAPI feature gate enables/disables processing of Kubernetes Gateway API CRDs. | false |
+| `FeatureGates.enableEndpointSlice` | EnableEndpointSlice feature gate enables/disables processing of Kubernetes EndpointSlices instead of Endpoints. | true |
 | `GatewayAPI.Image.repository` | Specify docker-registry that has the ako-gateway-api image | projects.registry.vmware.com/ako/ako-gateway-api:1.11.1 |
```
Comment (pkoshtavmware) [inline docs/ako_operator.md:127]: Not sure if this should be 1.11.1 or 1.13.1

Comment (pkoshtavmware) [reply docs/ako_operator.md:127]: cc @akshayhavile

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Review by aaha97 — APPROVED** [review]
LGTM

**Comment by aaha97** [comment]
build ako

**Pull request merged**